compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -132,8 +132,20 @@ class ClassfileParser(
     /** Parse parents for Java classes. For Scala, return AnyRef, since the real type will be unpickled.
      *  Updates the read pointer of 'in'. */
     def parseParents: List[Type] = {
-      val superType = if (isAnnotation) { in.nextChar; defn.AnnotationClass.typeRef }
-                      else pool.getSuperClass(in.nextChar).typeRef
+      val superType =
+        if (isAnnotation) {
+          in.nextChar
+          defn.AnnotationClass.typeRef
+        }
+        else if (classRoot.symbol == defn.ComparableClass ||
+                 classRoot.symbol == defn.JavaCloneableClass ||
+                 classRoot.symbol == defn.JavaSerializableClass) {
```

Comment:
You can do `eq` on a value whose type is a Java interface, but `eq` is not a member of `Any`. Here I'm just reproducing scalac's behavior to avoid inconsistencies.